libtiledbsoma/src/soma/soma_array.h

```diff
@@ -619,6 +619,14 @@ class SOMAArray : public SOMAObject {
      */
     void resize(const std::vector<int64_t>& newshape);
 
+    /**
+     * @brief Given an old-style array without current domain, sets its
+     * current domain. This is applicable only to arrays having all dims
+     * of int64 type. Namely, all SparseNDArray/DenseNDArray, and
+     * default-indexed DataFrame.
+     */
+    void upgrade_shape(const std::vector<int64_t>& newshape);
```

Comment:
`set_shape`? `upgrade_with_current_domain`? `upgrade_shape` sounds like what `resize` does where there's already a shape set, but you're updating with a new shape but maybe that's just me.

Reply:
Sorry but `upgrade_shape` was agree on by committee -- see #2407 which contains a copy/paste of our Google doc

This many months in I don't want to relitigate design ..... just wanting to move forward with what the group decision was.

Reply:
> ... upgrade_shape sounds like what resize does where there's already a shape set, but you're updating with a new shape but maybe that's just me.

Indeed it's almost identical code -- `upgrade_shape` throws if there is already a current domain; `resize` throws if there isn't; and then everything else is the same.

I thought about leaving it just `resize` at the C++ level, but decided to keep the names separate at the C++ level to mirror the committee-decided name-separation at the user API level ...